Dockerfile

```diff
@@ -21,7 +21,13 @@ RUN curl -o nmc-aarch64 -L https://github.com/suse-edge/nm-configurator/releases
     chmod +x nmc-aarch64 && \
     curl -o nmc-x86_64 -L https://github.com/suse-edge/nm-configurator/releases/download/v0.2.0/nmc-linux-x86_64 && \
     chmod +x nmc-x86_64 && \
-    cp nmc-$(uname -m) /usr/local/bin/nmc
+    cp nmc-$(uname -m) /usr/local/bin/nmc && \
+    curl -o hauler-amd64.tar -L https://github.com/rancherfederal/hauler/releases/download/v0.4.2/hauler_0.4.2_linux_amd64.tar.gz && \
+    tar -xf hauler-amd64.tar && \
+    mv hauler /usr/bin/hauler-amd64 && \
+    curl -o hauler-arm64.tar -L https://github.com/rancherfederal/hauler/releases/download/v0.4.2/hauler_0.4.2_linux_arm64.tar.gz && \
+    tar -xf hauler-arm64.tar && \
+    mv hauler /usr/bin/hauler-arm64
```

Comment:
I'd suggest to move this to a separate layer since it's different from the network configurations and the thought of constantly adding new dependencies and invalidating existing layers seems off since this can grow exponentially. I know @jdob is not generally a fan of many layers though. Thoughts?

Reply:
I actually had the same thought as Atanas when I saw this. I think layers being separated by functionality makes sense.